### PR TITLE
Bugfix for toolbox autopopulate

### PR DESCRIPTION
**Pull request by balderdash**
Stop adding draw_(turn|move)_by_constant_dropdown blocks to the toolbox when
they're already there.

### CODE REVIEW

File: dashboard/app/models/artist.rb [hunk 1]
```diff
@@ -284,17 +284,21 @@ def strip_block(block, toolbox_block, create_for_toolbox)
...
   end
 
   def create_toolbox_block(block)
     strip_block block, false, true
   end
 
+  def blocks_match?(toolbox_block, solution_block)
+    stripped_solution_block = strip_block(solution_block)
+    # Return true for either an exact match, or a *_constant block in the
+    # solution with the corresponding *_constant_dropdown block in the toolbox
```
Comment (Hamms) [inline dashboard/app/models/artist.rb]: presumably this actually returns true for a `*` block in the solution with a corresponding `*_dropdown` block in the toolbox; turn/move by constants are not the only blocks with a dropdown variant

**Comment by Hamms** [comment]
It might be worthwhile to generalize the tests and documentation a bit to make it clear that this distinction is true for all dropdown variants. Something like `controls_repeat_simplified` and `controls_repeat_simplified_dropdown`, for example

**Comment by balderdash** [comment]
Updated comments and tests.

**Comment by Hamms** [comment]
looks great!

**Pull request merged**